src/main/java/com/facebook/openwifirrm/optimizers/channel/RandomChannelInitializer.java

```diff
@@ -38,11 +38,38 @@ public class RandomChannelInitializer extends ChannelOptimizer {
 	/** The RRM algorithm ID. */
 	public static final String ALGORITHM_ID = "random";
 
+	/**
+	 * Args key for setting different channel per AP
+	 */
+	public static final String ARG_SET_DIFFERENT_CHANNEL_PER_AP =
+		"setDifferentChannelPerAp";
+
 	/** The PRNG instance. */
 	private final Random rng;
 
 	/** Whether to set a different value per AP or use a single value for all APs */
-	private final boolean setDifferentChannelsPerAp;
+	private final boolean setDifferentChannelPerAp;
+
+	/** Factory method to parse generic args map into the proper constructor */
+	public static RandomChannelInitializer makeWithArgs(
+		DataModel model,
+		String zone,
+		DeviceDataManager deviceDataManager,
+		Map<String, String> args
+	) {
+		if (args.containsKey(ARG_SET_DIFFERENT_CHANNEL_PER_AP)) {
+			boolean setDifferentChannelPerAp =
+				args.get(ARG_SET_DIFFERENT_CHANNEL_PER_AP) == "true";
```

Comment:
- Use String `.equals()`, this is Java 😛 
- I mildly prefer if you don't define class constants for the arg names, and just write them inline in this method to keep it all in one place. I don't suspect they'll be needed elsewhere.
- Avoid double lookups in maps.
- To make it easier to identify default values, I suggest avoiding the if-else blocks to call different constructors for their defaults, and instead define the defaults explicitly.

Suggestion:
```java
// Declare args with defaults at the top
boolean setDifferentChannelPerAp = false;

// Parse and validate args
String arg;
if ((arg = args.get("setDifferentChannelPerAp")) != null) {
  if (arg.equals("true")) {
    setDifferentChannelPerAp = true;
  }
}
if ((arg = args.get("something_else")) != null) {
  /* ... */
}

// Use one constructor only
return new <...>;
```

Reply:
Also note you can use `Boolean.parseBoolean()`